### PR TITLE
Integrate sliding window limiter into exchange backup operation

### PR DESCRIPTION
**Pull request by pandeyabs**
2nd last PR in the chain. Wires up the new sliding limiter into exchange backup create flows. I'll add some e2e regression tests to wrap it up.

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [x] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/operations/backup.go [hunk 1]
```diff
@@ -424,11 +425,20 @@ func (op *BackupOperation) do(
...
 	}
 
+	// Use sliding window limiter for Exchange if the feature is not explicitly
+	// disabled. For other services we don't use sliding window limiter.
+	enableSlidingLim := false
+	if service == path.ExchangeService &&
+		!op.Options.ToggleFeatures.DisableSlidingWindowLimiter {
+		enableSlidingLim = true
+	}
+
```
Comment (ryanfkeepers) [inline src/internal/operations/backup.go:435]: We're getting into bad habits here.  The operation layer should not investigate the service being handled.  These sorts of conditionals need to be handled via an interface that's common to all backup providers.  Something like `ctx = op.backupProducer.SetRateLimiter(ctx, op.Selector, op.Options)`.

**Review by ryanfkeepers — CHANGES_REQUESTED** [review]

File: src/internal/operations/backup.go (hunk 1 above)
Comment (pandeyabs) [reply src/internal/operations/backup.go:435]: Good point! I'll make the change.

**Comment by pandeyabs** [comment]
@ryanfkeepers this is ready for a final review. Thank you!

File: src/internal/operations/backup.go [hunk 2]
```diff
@@ -326,6 +322,12 @@ func (op *BackupOperation) Run(ctx context.Context) (err error) {
 		logger.Ctx(ctx).Infow("completed backup", "results", op.Results)
 	}
 
+	// For exchange, rate limits are enforced on a mailbox level. Reset the
+	// rate limiter so that it doesn't accidentally throttle following mailboxes.
+	// This is a no-op if we are using token bucket limiter since it refreshes
+	// tokens on a fixed per second basis.
+	graph.ResetLimiter(ctx)
```
Comment (ryanfkeepers) [inline src/internal/operations/backup.go:329]: ctx aren't mutable, you're slicing out the changes:
```suggestion
	ctx = graph.ResetLimiter(ctx)
```

File: src/pkg/services/m365/api/graph/concurrency_middleware.go [hunk 3]
```diff
@@ -193,6 +223,11 @@ func QueueRequest(ctx context.Context) {
 	}
 }
 
+func ResetLimiter(ctx context.Context) {
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/api/graph/concurrency_middleware.go:226]: Can you add a comment explaining how this affects the limiter, and when it should reasonably be used?

**Review by ryanfkeepers — APPROVED** [review]
You have a bug, but otherwise things look good.  Thanks!

**Pull request merged**